### PR TITLE
oop_principles / it-academy.by / JC1-110-22 / Anastasia Melnikova/ Done tasks 1-2

### PR DESCRIPTION
**Pull request by Stellmaria**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [x] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Vampire.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     * @param hero type hero.
+     */
+    @Override
+    protected void attackEnemy(Hero hero) {
+        hero.takingDamage(DAMAGE_VAMPIRE);
+    }
+
+    public int getDAMAGE_VAMPIRE() {
+        return DAMAGE_VAMPIRE;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Vampire.java:44]: Remove it

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Werewolf.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        }
+    }
+
+    public int getDAMAGE_WEREWOLF() {
+        return DAMAGE_WEREWOLF;
+    }
+
+    public int getDAMAGE_WEREWOLF_NIGHT() {
+        return DAMAGE_WEREWOLF_NIGHT;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Werewolf.java:91]: remove it

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     *
+     * @param damage damage taken.
+     */
+    public void takingDamage(int damage) {
+        if (isAlive()) {
+            if (getHealth() - damage < 0) {
+                setHealth(0);
+            } else if (getHealth() - damage == 0) {
+                boolean live = random.nextBoolean();
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java:38]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                if (live) {
+                    setHealth(super.getHealth());
+                } else {
+                    setHealth(getHealth() - damage);
+                }
+            } else {
+                setHealth(getHealth() - damage);
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            }
+        }
+    }
+
+    public int getDAMAGE_ZOMBIE() {
+        return DAMAGE_ZOMBIE;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java:52]: remove it

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/cats_home/Sphynx.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+     * @param user person type object.
+     * @return meow text.
+     */
+    @Override
+    protected String mew(Person user) {
+        final double NEGATIVE_PERCENTAGE = -10.0;
+
+        user.changeHappiness(NEGATIVE_PERCENTAGE);
+
+        return "meow";
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/cats_home/Sphynx.java:33]: reuse method `mew()`

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/EnemyTest.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.util.stream.Stream;
+
+public class EnemyTest {
+    final Enemy enemy = new Enemy(0) {
+        @Override
+        protected void attackEnemy(Hero hero) {
+            hero.takingDamage(0);
+        }
+    };
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/EnemyTest.java]: move to @BeforeAll

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/ZombieResurrectionTest.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
+package com.rakovets.course.java.core.practice.oop_principles.progect.battle_ground.enemy;
+
+import com.rakovets.course.java.core.practice.oop_principles.progect.battle_ground.hero.Archer;
+
+public class ZombieResurrectionTest {
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/ZombieResurrectionTest.java:7]:
```suggestion
public class ZombieResurrectionDemo {
```

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java (hunk 3 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java (hunk 4 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/cats_home/Sphynx.java (hunk 6 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/cats_home/Sphynx.java]: fixed

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/EnemyTest.java (hunk 7 above)
Comment (Stellmaria) [reply src/test/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/EnemyTest.java]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java (hunk 5 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/oop_principles/progect/battle_ground/enemy/Zombie.java]: fixed

**Pull request closed without merging**